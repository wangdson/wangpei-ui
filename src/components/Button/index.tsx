import React, { ButtonHTMLAttributes, PropsWithChildren, ReactNode, AnchorHTMLAttributes, useMemo } from "react";
import { Text, StyledButton, Loading } from './button';
import { SizeTypes, AppearancesTypes, APPEARANCES, SIZES } from "./types";
// @ts-ignore
export { SizeTypes, AppearancesTypes, APPEARANCES, SIZES };

export interface CustomButtonProps {
  /** 是否禁用 */
  disabled?: boolean;
  /** 是否加载中 */
  isLoading?: boolean;
  /** 是否是a标签 */
  isLink?: boolean;
  /** 按钮文本  */
  loadingText?: ReactNode;
  /** 按钮大小 */
  size?: SizeTypes;
  /** 按钮类型 */
  appearance?: AppearancesTypes;
  /** 无效点击 */
  isUnclickable?: boolean;
  /** click事件 */
  onClick?: ()=>void;
  /** 包含的icon */
  containsIcon?: boolean;
}

export type ButtonProps = CustomButtonProps & AnchorHTMLAttributes<HTMLAnchorElement> & ButtonHTMLAttributes<HTMLButtonElement>;
const Button: React.FC<ButtonProps> = (props: PropsWithChildren<ButtonProps>) => {
  const { isLoading, loadingText, isLink, children } = props;
  const buttonInner = (
    <>
      <Text>{children}</Text>
      {isLoading && <Loading>{loadingText || 'loading'}</Loading>}
    </>
  );
  const btnType = useMemo(()=>{
    if(isLink){
      return 'a'
    }
  }, [isLink]);
  return (
    <StyledButton as={btnType} {...props} data-testid="button">
      {buttonInner}
    </StyledButton>
  )
}

Button.defaultProps = {
  isLoading: false,
  loadingText: null,
  isLink: false,
  appearance: APPEARANCES.tertiary,
  disabled: false,
  isUnclickable: false,
  containsIcon: false,
  size: SIZES.medium
}

export default Button;