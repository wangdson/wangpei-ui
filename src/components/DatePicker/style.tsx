import styled, { css } from 'styled-components';
import { modalCloseAnimate, modalOpenAnimate } from '../shared/animation';
import { color, typography } from "../shared/style";
import { DateItem } from './index';

export const CalendarWrapper = styled.div<{visible: boolean, delay: number}> `
  position: absolute;
  border: 1px solid black;
  transition: all ${(props) => props.delay / 1000 }s cubic-bezier(0.23, 1, 0.32, 1);
  ${(props) => props.visible && css`animation: ${modalOpenAnimate} ${props.delay/1000}s ease-in;`  }
  ${(props) => !props.visible && css`animation: ${modalCloseAnimate} ${props.delay/1000}s ease-in;` }
`;

export const CalendarDateRow = styled.tr``;

export const tableItemStyle = css`
	display: inline-block;
	min-width: 24px;
	height: 24px;
	line-height: 24px;
	border-radius: 2px;
	margin: 2px;
	text-align: center;
`;

export const CalendarHeadItem = styled.td`
  ${tableItemStyle}
  cursor:default;
`;

export const CalendarHeadWrapper = styled.div`
  padding: 10px;
  display: flex;
  color: rgba(0, 0, 0, 0.85);
  border-bottom: 1px solid #f0f0f0;
  justify-content: center;
`;

export const btnStyle = {
	padding: "0px",
	background: color.lightest,
};

export const IconWrapper = styled.span`
  display: inline-block;
  vertical-align: middle;
  > svg {
    height: 12px;
  }
`;

export const BtnDiv = styled.div`
  display: flex;
  jutify-content: center;
  align-items: center;
  height: 24px;
  line-height: 24px;
`;

export const CalendarDate = styled.td<Partial<DateItem>>`
  display: inline-block;
  min-width: 24px;
  height: 24px;
  line-height: 24px;
  border-radius: 2px;
  margin: 2px;
  text-align: center;
  cursor: pointer;
  ${(props) => {
    if (props.isToday) {
      return `color:${color.lightest};background:${color.secondary};`;
    }
  }}
  ${(props) => {
    if (props.isonDay) {
      //当天的 
      return `color:${color.lightest};background:${color.primary};`;
    }
    return `&:hover {color: ${color.secondary};};`;
  }}
  ${(props) => {
    if (props.isonMonth === false) {
      //不是当月显示灰色
      return `color:${color.mediumdark};`;
    }
    return "";
  }}
`;

export const MonthWrapper = styled.div`
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  position: relative;
`;

export const MonthItem = styled.div<{ toGrey?: boolean }>`
  width: 25%;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  ${(props) => props.toGrey && `color:${color.mediumdark};`}
  &:hover {
    color: ${color.secondary};
  }
`;

export const Bwrapper = styled.b`
  cursor: pointer;
  &:hover {
    color: ${color.primary};
  }
`;

export const CalendarIcon = styled.span`
  display: inline-block;
`; 

export const DatePickerWrapper = styled.div`
  display: inline-block;
  border-color: #40a9ff;
  border-right-width: 1px !important;
  outline: 0;
  box-shadow: 0 0 0 2px rgba(24, 144, 255, 0.2);
`;
