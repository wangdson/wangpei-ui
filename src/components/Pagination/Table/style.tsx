import styled from 'styled-components';

export const TableTable = styled.table`
width: 100%;
text-align: left;
border-radius: 2px 2px 0 0;
border-collapse: separate;
border-spacing: 0;
table-layout: auto;
box-sizing: border-box;
margin: 0;
padding: 0;
color: rgba(0, 0, 0, 0.65);
font-variant: tabular-nums;
line-height: 1.5715;
list-style: none;
font-feature-settings: "tnum";
position: relative;
z-index: 0;
clear: both;
font-size: 14px;
background: #fff;
border-radius: 2px;
& > thead > tr > th {
  color: rgba(0, 0, 0, 0.85);
  font-weight: 500;
  text-align: left;
  background: #fafafa;
  border-bottom: 1px solid #f0f0f0;
  transition: background 0.3s ease;
  position: relative;
  padding: 16px;
  overflow-wrap: break-word;
}
& > tbody > tr {
  & > td {
    border-bottom: 1px solid #f0f0f0;
    transition: background 0.3s;
    position: relative;
    padding: 16px;
    overflow-wrap: break-word;
  }
  &:hover {
    & > td {
      background: #fafafa;
    }
  }
}
`;

export const TableHeadSpan = styled.span`
  display: inline-block;
  position: absolute;
  right: 0;
  top: 0;
  padding: 4px 16px 16px;
  cursor: pointer;
  & svg {
    height: 10px;
    width: 10px;
  }
`;