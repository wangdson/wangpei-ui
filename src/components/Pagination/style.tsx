import styled from 'styled-components';

export const PageUl = styled.ul`
  display: flex;
  justify-content: center;
  align-items: center;

  & > li {
    list-style: none;
    margin: 4px;
  }
  & button {
    border-radius: 6px;
    padding: 10px 8px;
  }
  & span {
    line-height: 13.6px;
    height: 13.6px;
    min-width: 18px;
  }
  & svg {
    height: 13.6px;
    width: 13.6px;
    vertical-align: bottom;
  }
`;