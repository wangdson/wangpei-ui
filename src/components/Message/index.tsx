import React, { PropsWithChildren, ReactNode, useState, useMemo, useEffect } from 'react';
import styled from 'styled-components';
import { iconSpin } from "../shared/animation";
import { darken, rgba, opacify } from "polished";
import { easing } from "../shared/animation";
import ReactDom from 'react-dom';
import { messageBoxShadow, IconWrapper, MessageTextWrapper, MessageText } from './style';
import { color } from '../shared/style';
import { Icon } from '../Icon';

export type MessageType =
	| "info"
	| "success"
	| "error"
	| "warning"
	| "loading"
  | "default";
  
export interface MessageConfig {
  /** 挂载点 */
  mount: HTMLElement;
  /** 动画延迟时间 */
  delay: number;
  /** 结束后回调 */
  callback: any;
  /** 动画持续时间 */
  animationDuring: number;
  /** 底色 */
  background: string;
  /** 文本颜色 */
  color: string;
}

/** 默认配置 */
const defaultConfig: MessageConfig = {
  mount: document.body,
  delay: 2000,
  callback: null,
  animationDuring: 300,
  background: color.lightest,
  color: color.dark,
}  

let wrap: HTMLElement;
export const createMessage = (type: MessageType) => {
	return (content: ReactNode, config: Partial<MessageConfig> = {}) => {
    const fconfig = {...defaultConfig, ...config};
    if (!wrap) {
			//如果有的话，说明已经调用过这个函数了，这个空div就可以一直复用
			wrap = document.createElement("div");
			wrap.style.cssText = `line-height:
        1.5;text-align:
        center;color: #333;
        box-sizing: border-box;
        margin: 0;
        padding: 0;
        list-style: none;
        position: fixed;
        z-index: 100000;
        width: 100%;
        top: 16px;
        left: 0;
        pointer-events: none;`;
      if (wrap) {
        fconfig.mount.appendChild(wrap); //挂body上
      }
		}

		const divs = document.createElement("div");
		wrap.appendChild(divs);
		ReactDom.render(
			<Message rootDom={wrap} parentDom={divs} content={content} fconfig={fconfig} iconType={type}/>,
			divs
		);
	};
};

export interface MessageProps {
  rootDom: HTMLElement; //这个用来干掉parentDom 这个可以常驻
	parentDom: Element | DocumentFragment; //这个是挂载点 要unmount卸载 完毕后卸载挂载点 注意！一共2步卸载，别漏了
  content: ReactNode;
  fconfig: MessageConfig;
  iconType: MessageType;
}

export function Message(props: PropsWithChildren<MessageProps>){
  const { rootDom, parentDom, content, fconfig, iconType } = props;
  const [close, setClose] = useState(false);

  // icon的render
  const renderIcon = useMemo(()=>{
    switch(iconType) {
      case 'default':
        return null;
      case "info":
        return (
          <IconWrapper>
            <Icon icon="info" color={color.primary}></Icon>
          </IconWrapper>
        );
      case 'success':
        return (
          <IconWrapper>
            <Icon icon="check" color={color.positive}></Icon>
          </IconWrapper>
        ); 
      case 'warning':
        return (
          <IconWrapper>
            <Icon icon="info" color={color.warning}></Icon>
          </IconWrapper>
        ); 
      case 'error':
        return (
          <IconWrapper>
            <Icon icon="closeAlt" color={color.negative}></Icon>
          </IconWrapper>
        );   
      case 'loading':
        return (<IconWrapper spin={true}>
          <Icon icon="sync"></Icon>
        </IconWrapper>);
      default:
        return null;  
    }
  }, [ iconType ]);
  const unmount = useMemo(()=>{
    // 解绑函数
    return ()=>{
      if (parentDom && rootDom) {
        ReactDom.unmountComponentAtNode(parentDom);
        rootDom.removeChild(parentDom);
      }
    }
  },[parentDom, rootDom]);
  
  useEffect(()=>{
    // 结束操作
    // 计算延迟和动画延迟的时间差
    let closeStart = fconfig.delay - fconfig.animationDuring;
    // 先打开
    let timer1 = window.setTimeout(()=>{setClose(true)}, closeStart > 0? closeStart: 0);
    // 再关闭操作
    let timer2 = window.setTimeout(()=>{
      setClose(false);
      unmount();
      if(fconfig.callback) {
        fconfig.callback();
      }
    }, fconfig.delay);
    return ()=>{
      window.clearTimeout(timer1);
      window.clearTimeout(timer2);
    }
  }, [unmount,fconfig]);

  // 监测unmount的变化，变化则重新调用
  // useEffect(()=>{
  //   setTimeout(() => {
  //     unmount();
  //   }, 2000);
  // }, [unmount]);
  return (
    <MessageTextWrapper openState={true} closeState={close} ani={fconfig.animationDuring}>
      <MessageText bg={fconfig.background} fc={fconfig.color}>
        {renderIcon}
        {content}
      </MessageText>
    </MessageTextWrapper>
  )
}

export const message = {
	info: createMessage("info"),
	success: createMessage("success"),
	error: createMessage("error"),
	warning: createMessage("warning"),
	loading: createMessage("loading"),
	default: createMessage("default"),
};