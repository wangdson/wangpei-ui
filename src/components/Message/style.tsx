import styled, {css} from 'styled-components';
import { typography } from "../shared/style";
import { iconSpin, messageOpenAnimate, messageCloseAnimate } from '../shared/animation';

export const messageBoxShadow = css`
	box-shadow: 0 3px 6px -4px rgba(0, 0, 0, 0.12),
		0 6px 16px 0 rgba(0, 0, 0, 0.08), 0 9px 28px 8px rgba(0, 0, 0, 0.05);
`;

export const MessageText = styled.span<{ bg: string; fc: string }>`
	display: inline-block;
	padding: 10px 16px;
	background: ${(props) => props.bg};
	color: ${(props) => props.fc};
	font-size: ${typography.size.s2}px;
	font-weight: ${typography.weight.bold};
	margin: 10px;
	${messageBoxShadow};
	border-radius: 2px;
`;

export const IconWrapper = styled.span<{ spin?: boolean }>`
	margin-right: 10px;
	& > svg {
		font-size: ${typography.size.s2}px;
		${(props) =>
			props.spin &&
			css`
				animation: ${iconSpin} 2s linear infinite;
			`}
	}
`;

export const MessageTextWrapper = styled.div<{
	openState: boolean;
	closeState: boolean;
	ani: number;
}>`
	${(props) =>
		props.openState &&
		css`
			animation: ${messageOpenAnimate} ${props.ani / 1000}s ease-in;
		`}
	${(props) =>
		props.closeState &&
		css`
			animation: ${messageCloseAnimate} ${props.ani / 1000}s ease-in;
		`}
`;