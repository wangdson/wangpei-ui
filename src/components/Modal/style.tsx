import styled, {css} from 'styled-components';
import {typography} from '../../components/shared/style'; 
import { modalOpenAnimate, modalCloseAnimate } from '../../components/shared/animation'; 

export const ModalWrapper = styled.div`
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	z-index: 1000;
`;

export const ModalViewPort = styled.div<{ visible: boolean; delay: number }>`
background: #fff;
border: none;
border-radius: 5px;
box-shadow: 2px 2px 4px #d9d9d9;
text-shadow: 1px 1px 4px #d9d9d9, -1px -1px 4px #fff;
margin: 0 auto;
min-width: 320px;
overflow: hidden;
padding: 8px;
position: relative;
top: 100px;
transition: all 0.5s cubic-bezier(0.23, 1, 0.32, 1);
width: 30%;
z-index: 1001;
${(props) =>
  props.visible &&
  css`
    animation: ${modalOpenAnimate} ${props.delay / 1000}s ease-in;
  `}
${(props) =>
  !props.visible &&
  css`
    animation: ${modalCloseAnimate} ${props.delay / 1000}s ease-in;
  `}
`;

export const ModalMask = styled.div`
  background-color: rgba(0, 0, 0, 0.45);
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
`;

export const CloseBtn = styled.div`
  position: absolute;
  right: 5px;
  top: 5px;
`;

export const ConfirmWrapper = styled.div`
	display: flex;
	justify-content: flex-end;
	align-item: center;
	padding: 10px;
`;

export const ChildrenWrapper = styled.div`
	min-height: 100px;
	padding: 10px;
`;

export const TitleWrapper = styled.div`
  height: 30px;
  line-height: 30px;
  font-size: ${typography.size.m2}px;
  font-weight: ${typography.weight.bold};
  padding: 5px;
`;