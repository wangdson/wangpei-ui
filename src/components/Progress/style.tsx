import styled,{keyframes} from 'styled-components';
import {typography} from '../shared/style';

export const progressFlash = keyframes`
  0% {
    opacity: 0.1;
    width: 0;
  }
  20% {
    opacity: 0.5;
    width: 0;
  }
  100% {
    opacity: 0;
    width: 100%
  }
`;

export const BarWrapper = styled.div`
  display: flex;
  padding: 5px;
  align-items: center;
`; 

export interface BarMainProps {
	state: number;
	height?: number;
	flashColor: string;
	primary: string;
	secondary: string;
}

export const BarMain = styled.div<BarMainProps>`
  width: ${(props) => props.state}%;
  height: ${(props) => (props.height ? props.height : 8)}px;
  background-color: ${(props) => props.primary};
  background-image: linear-gradient(
    to right,
    ${(props) => props.primary},
    ${(props) => props.secondary}
  );
  transition: all 0.4s cubic-bezier(0.08, 0.82, 0.17, 1) 0s;
  border-radius: 5px;
  position: relative;
  &::before {
    animation: ${progressFlash} 2.4s cubic-bezier(0.23, 1, 0.32, 1) infinite;
    background: ${(props) => props.flashColor};
    border-radius: 10px;
    bottom: 0;
    content: "";
    left: 0;
    opacity: 0;
    position: absolute;
    right: 0;
    top: 0;
  }
`;

export const BarMainWrapper = styled.div<{ bottomColor: string; height?: number }>`
	width: 100%;
	border-radius: 5px;
	position: relative;
	background: ${(props) => props.bottomColor};
	height: ${(props) => (props.height ? props.height : 8)}px;
`;
export const BarText = styled.div<{ height?: number }>`
	line-height: ${(props) => (props.height ? props.height : 8)}px;
	font-weight: ${typography.weight.bold};
	text-align: center;
	display: inline-block;
	margin-left: 10px;
	min-width: 55px;
`;
export const CircleWrapper = styled.div`
	position: relative;
	display: inline-block;
	border-radius: 50%;
`;

export const CircleText = styled.div<{ size: number }>`
	line-height: ${(props) => props.size * 0.62}px;
	width: ${(props) => props.size * 0.62}px;
	height: ${(props) => props.size * 0.62}px;
	border-radius: 50%;
	display: inline-block;
	font-weight: ${typography.weight.bold};
	left: 50%;
	position: absolute;
	text-align: center;
	top: 50%;
	transform: translateX(-50%) translateY(-50%);
`;


