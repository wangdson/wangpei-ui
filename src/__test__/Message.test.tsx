import React from "react";
import { act, fireEvent, cleanup } from "@testing-library/react";
import { Message, MessageType, message, createMessage } from "../components/Message";
import { color, typography } from "../components/shared/style";
import { unmountComponentAtNode, render } from "react-dom";
import Button from '../components/Button';

let container: HTMLDivElement;
beforeEach(() => {
	// 创建一个 DOM 元素作为渲染目标
	container = document.createElement("div");
	document.body.appendChild(container);
});
const clean = () => {
	// 退出时进行清理
	unmountComponentAtNode(container);
};
afterEach(() => clean());
const sleep = (delay: number) => {
	return new Promise((res) => {
		setTimeout(() => {
			res();
		}, delay);
	});
};

async function changeIcon(type: MessageType) {
  clean();
  act(()=>{
    render(
			<Button
				id="btn"
				onClick={() => message[type](<span className="test3">test3</span>)}
			>
				but
			</Button>,
			container
		);
  })
  const btn = container.querySelector("#btn");
  await act(async () => {
		fireEvent.click(btn!);
		await sleep(500);
  });
  expect(document.querySelector('.test3')).toBeTruthy();
  expect(container).toMatchSnapshot();
}

const fn = jest.fn();

describe('test Message component', ()=>{
  it('render basic func', async ()=>{
    act(()=>{
      render(
        <Button
          id="btn"
          onClick={() => message.default(<span className="test">test</span>)}
        >
          but
        </Button>,
        container
      );
    })
    let btn = container.querySelector("#btn");
    await act(async ()=>{
      fireEvent.click(btn!);
      await sleep(500);
    })
    expect(document.querySelector('.test')).toBeTruthy();
    expect(container).toMatchSnapshot();
    await act(async ()=>{
      await sleep(1500);
    })
    expect(document.querySelector('.test')).toBeNull();
    expect(container).toMatchSnapshot();
  });
  it('change color', async ()=>{
    act(() => {
			render(
				<Button
					id="btn"
					onClick={() =>
						message.default(<span className="test2">test2</span>, {
							background: "blue",
							color: "red",
						})
					}
				>
					but2
				</Button>,
				container
			);
    });
    let btn = container.querySelector("#btn");
    await act(async ()=>{
      fireEvent.click(btn!);
      await sleep(500);
    })
    expect(document.querySelector(".test2")).toBeTruthy();
		expect(container).toMatchSnapshot();
  });
  it('callback test', async ()=>{
    act(() => {
			render(
				<Button
					id="btn"
					onClick={() =>
						message.default(<span className="callback">callback</span>, {
							callback: fn
						})
					}
				>
					callbackBtn
				</Button>,
				container
			);
    });
    const btn = container.querySelector("#btn");
    expect(fn).not.toHaveBeenCalled();
    await act(async()=>{
      fireEvent.click(btn!);
      await sleep(2100);
    })
    expect(fn).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
  it('change icon', async ()=>{
    await changeIcon('default');
    await changeIcon('error');
    await changeIcon('success');
    await changeIcon('warning');
    await changeIcon('success');
    await changeIcon('loading');
  });
  //prettier-ignore
	it("icon type default", async () => {
		act(() => {
			render(
				<div>
					{/* @ts-ignore */}
					<Button id="btn" onClick={() => createMessage("sdss")("fff")}
					>
						btn
					</Button>
				</div>,
				container
			);
		});
		const btn = container.querySelector("#btn");
		await act(async () => {
			fireEvent.click(btn!);
			await sleep(2100);
		});
		expect(container).toMatchSnapshot();
  });
  
  it("animate duration", async () => {
		act(() => {
			render(
				<Button
					id="btn"
					onClick={() =>
						message.default(<span className="callback">callback</span>, {
							animationDuring: 1000,
							delay: 3000,
						})
					}
				>
					btn
				</Button>,
				container
			);
		});
		let btn = container.querySelector("#btn");
		await act(async () => {
			fireEvent.click(btn!);
			await sleep(2100);
		});
		expect(container).toMatchSnapshot();
		await act(async () => {
			await sleep(1100);
		});
		expect(container).toMatchSnapshot();
  });
  it("animation lg delay", async () => {
		act(() => {
			render(
				<Button
					id="btn"
					onClick={() =>
						message.default(<span className="callback">callback</span>, {
							animationDuring: 2000,
							delay: 1000,
						})
					}
				>
					btn
				</Button>,
				container
			);
		});
		let btn = container.querySelector("#btn");
		await act(async () => {
			fireEvent.click(btn!);
			await sleep(1000);
		});
		expect(container).toMatchSnapshot();
	});
})