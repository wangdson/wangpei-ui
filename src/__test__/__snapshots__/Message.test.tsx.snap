// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Message component animate duration 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      btn
    </span>
  </button>
</div>
`;

exports[`test Message component animate duration 2`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      btn
    </span>
  </button>
</div>
`;

exports[`test Message component animation lg delay 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      btn
    </span>
  </button>
</div>
`;

exports[`test Message component callback test 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      callbackBtn
    </span>
  </button>
</div>
`;

exports[`test Message component change color 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but2
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 2`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 3`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 4`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 5`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component change icon 6`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component icon type default 1`] = `
<div>
  <div>
    <button
      class="sc-iCoHVE jZrQwz"
      data-testid="button"
      id="btn"
    >
      <span
        class="sc-jSFkmK dbViRT"
      >
        btn
      </span>
    </button>
  </div>
</div>
`;

exports[`test Message component render basic func 1`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;

exports[`test Message component render basic func 2`] = `
<div>
  <button
    class="sc-iCoHVE jZrQwz"
    data-testid="button"
    id="btn"
  >
    <span
      class="sc-jSFkmK dbViRT"
    >
      but
    </span>
  </button>
</div>
`;
